Library/readline/readline.c

```diff
@@ -0,0 +1,93 @@
+
+//
+// readline.c
+//
+// Copyright (c) 2014 Yorkie Neil <yorkiefixer@gmail.com>
+//
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "readline.h"
+
+readline_t *
+readline_new(char * buffer) {
+  readline_t * rl = (readline_t*) malloc(sizeof(readline_t));
+  size_t len = strlen(buffer);
+
+  rl->cursor = 0;
+  rl->line = 0;
+  rl->buffer = (char*) malloc(len);
+  memset(rl->buffer, 0, len);
```

Comment:
It is a dependency-break idiom in superscalar pipelined arches, unnecessary in our case. I will strip it in a later version.